### PR TITLE
Remove unbounded_input from FileSinkOptions

### PR DESCRIPTION
**Pull request by devinjdangelo**
## Which issue does this PR close?

None, but related to #8604 

## Rationale for this change

Now that writing unbounded (streaming) sources is supported via `StreamWrite`, other `DataSink` impls do not need to worry about yielding on unbounded input plans. 

## What changes are included in this PR?

Removes unbounded_input checks in FileSinkConfig

## Are these changes tested?

Yes, by existing tests

## Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/proto/gen/src/main.rs [hunk 1]
```diff
@@ -28,6 +28,7 @@ fn main() -> Result<(), String> {
     let descriptor_path = proto_dir.join("proto_descriptor.bin");
 
     prost_build::Config::new()
+        .protoc_arg("--experimental_allow_proto3_optional")
```
Comment (devinjdangelo) [inline datafusion/proto/gen/src/main.rs:31]: Should be able to remove this after https://github.com/apache/arrow-datafusion/issues/8602 is fixed

**Review by alamb — APPROVED** [review]
Thank you (again) @devinjdangelo 

@metesynnada  / @ozankabak  can you please also review this PR to make sure we are not removing something that is important for your usecase?

File: datafusion/proto/gen/src/main.rs (hunk 1 above)
Comment (devinjdangelo) [reply datafusion/proto/gen/src/main.rs]: Removed merged main, and confirmed that 8602 is fixed for me as well

**Review by metesynnada — APPROVED** [review]
LGTM.

**Comment by alamb** [comment]
🚀

**Pull request merged**